matching-core/src/main/java/com/graphhopper/matching/MatchEntry.java

```diff
@@ -1,30 +1,80 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for 
+ *  additional information regarding copyright ownership.
+ * 
+ *  GraphHopper GmbH licenses this file to you under the Apache License, 
+ *  Version 2.0 (the "License"); you may not use this file except in 
+ *  compliance with the License. You may obtain a copy of the License at
+ * 
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
 package com.graphhopper.matching;
 
-import java.util.List;
-
-import com.bmw.hmm.SequenceState;
-import com.graphhopper.matching.util.TimeStep;
-import com.graphhopper.routing.Path;
 import com.graphhopper.util.EdgeIteratorState;
 import com.graphhopper.util.GPXEntry;
 import com.graphhopper.util.shapes.GHPoint3D;
 
 public class MatchEntry {
-    public final int sequenceIdx;
+    public static enum MatchState { MATCHING_NOT_STARTED, NOT_USED_FOR_MATCHING, MATCHED };
+    private MatchState matchState = MatchState.MATCHING_NOT_STARTED; 
     public final GPXEntry gpxEntry;
-    public final List<GPXEntry> neighboringGpxEntries;
-    public final GHPoint3D point;
-    public final EdgeIteratorState directedRealEdge;
-    public final int sequenceMatchEdgeIdx;
-    public final double distanceAlongRealEdge;
-    public MatchEntry(int sequenceIdx, SequenceState<GPXExtension, GPXEntry, Path> matchStep, TimeStep timeStep, EdgeIteratorState directedRealEdge, int sequenceMatchEdgeIdx, double distanceAlongRealEdge) {
+    private int sequenceIdx;
+    private GHPoint3D snappedPoint;
+    private EdgeIteratorState directedRealEdge;
+    private int sequenceMatchEdgeIdx;
+    private double distanceAlongRealEdge;
+    private boolean locked = false;
+    
+    protected MatchEntry(GPXEntry gpxEntry) {
+        this.gpxEntry = gpxEntry;
+    }
+    
+    protected void markAsNotUsedForMatching() {
+        assert !locked;
+        this.matchState = MatchState.NOT_USED_FOR_MATCHING;
+        locked = true;
+    }
+    
+    protected void saveMatchingState(int sequenceIdx, int sequenceMatchEdgeIdx,
```

Comment:
In this case you should use [package-private](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html) (no modifier) instead of protected. Protected should only be used if you plan to access the method in derived classes.